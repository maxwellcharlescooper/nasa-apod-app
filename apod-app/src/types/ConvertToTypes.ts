import { Apod } from "./Apod";

export const toApodResponse = (response: any) => response as Apod[];
