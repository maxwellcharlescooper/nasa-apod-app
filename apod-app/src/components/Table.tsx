import React from "react";
import DataTable from "react-data-table-component";
import { Apod } from "../types/Apod";
import Colors from "../styles/Colors";
import { format, parse } from "date-fns";

const columns = [
  {
    name: "Title",
    selector: row => row.title,
    width: "20vw"
  },
  {
    name: "Date",
    selector: row =>
      format(parse(row.date, "yyyy-MM-dd", new Date()), "LLL d, yyyy"),
    width: "10vw"
  },
  {
    name: "Explanation",
    selector: row => row.explanation,
    width: "30vw"
  }
];

const customStyles = {
  headRow: {
    style: {
      backgroundColor: Colors.YELLOW
    }
  }
};

type Props = {
  data: Apod[];
  handleRowClick: (row, event) => void;
};

const Table = ({ data, handleRowClick }: Props) => {
  return (
    <DataTable
      columns={columns}
      data={data}
      highlightOnHover
      pointerOnHover
      customStyles={customStyles}
      onRowClicked={handleRowClick}
    />
  );
};

export default Table;
