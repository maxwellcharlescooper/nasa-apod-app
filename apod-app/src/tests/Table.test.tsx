import Table from "../components/Table";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import { Apod } from "../types/Apod";
import "@testing-library/jest-dom";
import { testData } from "../tests/testData";

describe("Table.tsx", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Table
          data={testData}
          handleRowClick={(row: Apod, event) => {
            return;
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders modal on click", () => {
    const handleClick = jest.fn();
    render(<Table data={testData} handleRowClick={handleClick} />);

    fireEvent.click(screen.getByText(/Jun 27, 2023/)); // trigger click event on the element

    expect(handleClick).toBeCalledTimes(1);
  });
});
