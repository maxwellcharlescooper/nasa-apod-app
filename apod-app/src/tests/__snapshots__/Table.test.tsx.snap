// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.tsx renders correctly 1`] = `
<div
  className="sc-dmlrgD epRaQS"
  fixedHeader={false}
  fixedHeaderScrollHeight="100vh"
  responsive={true}
>
  <div
    className="sc-fKFzqx bjJfGV"
  >
    <div
      className="sc-bdfCkl QpQhc rdt_Table"
      disabled={false}
      role="table"
    >
      <div
        className="sc-gsTChe fiBztD rdt_TableHead"
        fixedHeader={false}
        role="rowgroup"
      >
        <div
          className="sc-dlfmnT cUueSI rdt_TableHeadRow"
          dense={false}
          role="row"
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-iqHXTm cgcVre fLUulJ cImcsm rdt_TableCol"
            data-column-id={1}
            headCell={true}
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            width="20vw"
          >
            <div
              className="sc-crrsTd fpZkcf rdt_TableCol_Sortable"
              data-column-id={1}
              data-sort-id={1}
              disabled={true}
              role="columnheader"
              sortActive={false}
              tabIndex={0}
            >
              <div
                className="sc-dQqcE fyYZcz"
                data-column-id={1}
              >
                Title
              </div>
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-iqHXTm cgcVre OOsNt cImcsm rdt_TableCol"
            data-column-id={2}
            headCell={true}
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            width="10vw"
          >
            <div
              className="sc-crrsTd fpZkcf rdt_TableCol_Sortable"
              data-column-id={2}
              data-sort-id={2}
              disabled={true}
              role="columnheader"
              sortActive={false}
              tabIndex={0}
            >
              <div
                className="sc-dQqcE fyYZcz"
                data-column-id={2}
              >
                Date
              </div>
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-iqHXTm cgcVre dNPYEN cImcsm rdt_TableCol"
            data-column-id={3}
            headCell={true}
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            width="30vw"
          >
            <div
              className="sc-crrsTd fpZkcf rdt_TableCol_Sortable"
              data-column-id={3}
              data-sort-id={3}
              disabled={true}
              role="columnheader"
              sortActive={false}
              tabIndex={0}
            >
              <div
                className="sc-dQqcE fyYZcz"
                data-column-id={3}
              >
                Explanation
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-hHfsPW iotcTp rdt_TableBody"
        role="rowgroup"
      >
        <div
          className="sc-jrAHeI jijTP rdt_TableRow"
          dense={false}
          highlightOnHover={true}
          id="row-0"
          onClick={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          pointerOnHover={true}
          role="row"
          selected={false}
          striped={false}
          style={Object {}}
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn fLUulJ krQxxs rdt_TableCell"
            data-column-id={1}
            data-tag="allowRowEvents"
            id="cell-1-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="20vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              MAVEN's Ultraviolet Mars
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn OOsNt krQxxs rdt_TableCell"
            data-column-id={2}
            data-tag="allowRowEvents"
            id="cell-2-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="10vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Jun 27, 2023
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn dNPYEN krQxxs rdt_TableCell"
            data-column-id={3}
            data-tag="allowRowEvents"
            id="cell-3-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="30vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              These two global views of Mars were captured at ultraviolet wavelengths, beyond the spectrum visible to human eyes. Recorded by the MAVEN spacecraft's Imaging Ultraviolet Spectrograph instrument in July 2022 (left) and January 2023, three otherwise invisible ultraviolet bands are mapped into red, green, and blue colors. That color scheme presents the Red Planet's surface features in shades of tan and green. Haze and clouds appear white or blue, while high altitude ozone takes on a dramatic purple hue. On the left, Mars' south polar ice cap is in brilliant white at the bottom but shrinking during the southern hemisphere's summer season. On the right, the northern hemisphere's polar region is seen shrouded in clouds and atmospheric ozone. Known to some as the Mars Atmosphere and Volatile EvolutioN spacecraft, MAVEN has been exploring Mars' tenuous upper atmosphere, ionosphere, and its interactions with the Sun and solar wind since 2014.
            </div>
          </div>
        </div>
        <div
          className="sc-jrAHeI jijTP rdt_TableRow"
          dense={false}
          highlightOnHover={true}
          id="row-1"
          onClick={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          pointerOnHover={true}
          role="row"
          selected={false}
          striped={false}
          style={Object {}}
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn fLUulJ krQxxs rdt_TableCell"
            data-column-id={1}
            data-tag="allowRowEvents"
            id="cell-1-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="20vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Messier 24: Sagittarius Star Cloud
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn OOsNt krQxxs rdt_TableCell"
            data-column-id={2}
            data-tag="allowRowEvents"
            id="cell-2-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="10vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Jun 28, 2023
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn dNPYEN krQxxs rdt_TableCell"
            data-column-id={3}
            data-tag="allowRowEvents"
            id="cell-3-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="30vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Unlike most entries in Charles Messier's famous catalog of deep sky objects, M24 is not a bright galaxy, star cluster, or nebula. It's a gap in nearby, obscuring interstellar dust clouds that allows a view of the distant stars in the Sagittarius spiral arm of our Milky Way galaxy. Direct your gaze through this gap with binoculars or small telescope and you are looking through a window over 300 light-years wide at stars some 10,000 light-years or more from Earth. Sometimes called the Small Sagittarius Star Cloud, M24's luminous stars fill this gorgeous starscape.  Covering over 3 degrees or the width of 6 full moons in the constellation Sagittarius, the telescopic field of view includes dark markings B92 and B93 near center, along with other clouds of dust and glowing nebulae toward the center of the Milky Way.
            </div>
          </div>
        </div>
        <div
          className="sc-jrAHeI jijTP rdt_TableRow"
          dense={false}
          highlightOnHover={true}
          id="row-2"
          onClick={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          pointerOnHover={true}
          role="row"
          selected={false}
          striped={false}
          style={Object {}}
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn fLUulJ krQxxs rdt_TableCell"
            data-column-id={1}
            data-tag="allowRowEvents"
            id="cell-1-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="20vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              A Message from the Gravitational Universe
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn OOsNt krQxxs rdt_TableCell"
            data-column-id={2}
            data-tag="allowRowEvents"
            id="cell-2-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="10vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Jun 29, 2023
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn dNPYEN krQxxs rdt_TableCell"
            data-column-id={3}
            data-tag="allowRowEvents"
            id="cell-3-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="30vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Monitoring 68 pulsars with very large radio telescopes, the North American Nanohertz Observatory for Gravitational Waves (NANOGrav) has uncovered evidence for the gravitational wave (GW) background by carefully measuring slight shifts in the arrival times of pulses.  These shifts are correlated between different pulsars in a way that indicates that they are caused by GWs. This GW background is likely due to hundreds of thousands or even millions of supermassive black hole binaries.  Teams in Europe, Asia and Australia have also independently reported their results today. Previously, the LIGO and Virgo detectors have detected higher-frequency GWs from the merging of individual pairs of massive orbiting objects, such as stellar-mass black holes. The featured illustration highlights this spacetime-shaking result by depicting two orbiting supermassive black holes and several of the pulsars that would appear to have slight timing shifts.  The imprint these GWs make on spacetime itself is illustrated by a distorted grid.   Open Science: Browse 3,000+ codes in the Astrophysics Source Code Library
            </div>
          </div>
        </div>
        <div
          className="sc-jrAHeI jijTP rdt_TableRow"
          dense={false}
          highlightOnHover={true}
          id="row-3"
          onClick={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          pointerOnHover={true}
          role="row"
          selected={false}
          striped={false}
          style={Object {}}
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn fLUulJ krQxxs rdt_TableCell"
            data-column-id={1}
            data-tag="allowRowEvents"
            id="cell-1-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="20vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Orbits of Potentially Hazardous Asteroids
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn OOsNt krQxxs rdt_TableCell"
            data-column-id={2}
            data-tag="allowRowEvents"
            id="cell-2-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="10vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Jun 30, 2023
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn dNPYEN krQxxs rdt_TableCell"
            data-column-id={3}
            data-tag="allowRowEvents"
            id="cell-3-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="30vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Are asteroids dangerous? Some are, but the likelihood of a dangerous asteroid striking the Earth during any given year is low. Because some past mass extinction events have been linked to asteroid impacts, however, humanity has made it a priority to find and catalog those asteroids that may one day affect life on Earth. Pictured here are the orbits of the over 1,000 known Potentially Hazardous Asteroids (PHAs). These documented tumbling boulders of rock and ice are over 140 meters across and will pass within 7.5 million kilometers of Earth -- about 20 times the distance to the Moon. Although none of them will strike the Earth in the next 100 years -- not all PHAs have been discovered, and past 100 years, many orbits become hard to predict. Were an asteroid of this size to impact the Earth, it could raise dangerous tsunamis, for example. To investigate Earth-saving strategies, NASA successfully tested the Double Asteroid Redirection Test (DART) mission last year. Of course, rocks and ice bits of much smaller size strike the Earth every day, usually pose no danger, and sometimes create memorable fireball and meteor displays.    Today is: Asteroid Day
            </div>
          </div>
        </div>
        <div
          className="sc-jrAHeI jijTP rdt_TableRow"
          dense={false}
          highlightOnHover={true}
          id="row-4"
          onClick={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          pointerOnHover={true}
          role="row"
          selected={false}
          striped={false}
          style={Object {}}
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn fLUulJ krQxxs rdt_TableCell"
            data-column-id={1}
            data-tag="allowRowEvents"
            id="cell-1-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="20vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Three Galaxies in Draco
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn OOsNt krQxxs rdt_TableCell"
            data-column-id={2}
            data-tag="allowRowEvents"
            id="cell-2-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="10vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Jul 1, 2023
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn dNPYEN krQxxs rdt_TableCell"
            data-column-id={3}
            data-tag="allowRowEvents"
            id="cell-3-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="30vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              This tantalizing trio of galaxies sometimes called the Draco Group, is located in the northern constellation of (you guessed it) Draco, the Dragon. From left to right are face-on spiral NGC 5985, elliptical galaxy NGC 5982, and edge-on spiral NGC 5981, all found within this single telescopic field of view that spans a little more than the width of the full moon. While the group is far too small to be a galaxy cluster, and has not been catalogued as a compact galaxy group, the three galaxies all do lie roughly 100 million light-years from planet Earth. Not as well known as other tight groupings of galaxies, the contrast in visual appearance still makes this triplet an attractive subject for astroimagers. On close examination with spectrographs, the bright core of striking spiral NGC 5985 shows prominent emission in specific wavelengths of light, prompting astronomers to classify it as a Seyfert, a type of active galaxy. This impressively deep exposure hints at a faint dim halo along with sharp-edged shells surrounding elliptical NGC 5982, evidence of past galactic mergers. It also reveals many even more distant background galaxies.
            </div>
          </div>
        </div>
        <div
          className="sc-jrAHeI jijTP rdt_TableRow"
          dense={false}
          highlightOnHover={true}
          id="row-5"
          onClick={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          pointerOnHover={true}
          role="row"
          selected={false}
          striped={false}
          style={Object {}}
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn fLUulJ krQxxs rdt_TableCell"
            data-column-id={1}
            data-tag="allowRowEvents"
            id="cell-1-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="20vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Milky Way and Aurora over Antarctica
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn OOsNt krQxxs rdt_TableCell"
            data-column-id={2}
            data-tag="allowRowEvents"
            id="cell-2-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="10vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Jul 2, 2023
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn dNPYEN krQxxs rdt_TableCell"
            data-column-id={3}
            data-tag="allowRowEvents"
            id="cell-3-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="30vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              It was one of the better skies of this long night. In parts of Antarctica, not only is it winter, but the Sun can spend weeks below the horizon. At China's Zhongshan Station, people sometimes venture out into the cold to photograph a spectacular night sky. The featured image from one such outing was taken in mid-July of 2015, just before the end of this polar night. Pointing up, the wide angle lens captured not only the ground at the bottom, but at the top as well.  In the foreground, a colleague is taking pictures. In the distance, a spherical satellite receiver and several windmills are visible. Numerous stars dot the night sky, including Sirius and Canopus. Far in the background, stretching overhead from horizon to horizon, is the central band of our Milky Way Galaxy. Even further in the distance, visible as extended smudges near the top, are the Large and Small Magellanic Clouds, satellite galaxies near our huge Milky Way Galaxy.   Explore the Universe: Random APOD Generator
            </div>
          </div>
        </div>
        <div
          className="sc-jrAHeI jijTP rdt_TableRow"
          dense={false}
          highlightOnHover={true}
          id="row-6"
          onClick={[Function]}
          onDoubleClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          pointerOnHover={true}
          role="row"
          selected={false}
          striped={false}
          style={Object {}}
        >
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn fLUulJ krQxxs rdt_TableCell"
            data-column-id={1}
            data-tag="allowRowEvents"
            id="cell-1-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="20vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Venus in Ultraviolet from Akatsuki
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn OOsNt krQxxs rdt_TableCell"
            data-column-id={2}
            data-tag="allowRowEvents"
            id="cell-2-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="10vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Jul 3, 2023
            </div>
          </div>
          <div
            className="sc-hKgJyM sc-eCstFB sc-jSgtCu dsueRn dNPYEN krQxxs rdt_TableCell"
            data-column-id={3}
            data-tag="allowRowEvents"
            id="cell-3-undefined"
            isDragging={false}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            renderAsCell={false}
            role="cell"
            style={Object {}}
            width="30vw"
          >
            <div
              data-tag="allowRowEvents"
            >
              Why is Venus so different from Earth?  To help find out, Japan launched the robotic Akatsuki spacecraft which entered orbit around Venus late in 2015 after an unplanned five-year adventure around the inner Solar System.  Even though Akatsuki was past its original planned lifetime, the spacecraft and instruments were operating so well that much of its original mission was reinstated. Also known as the Venus Climate Orbiter, Akatsuki's instruments investigated unknowns about Earth's sister planet, including whether volcanoes are still active, whether lightning occurs in the dense atmosphere, and why wind speeds greatly exceed the planet's rotation speed. In the featured image taken by Akatsuki's UVI camera, the day-side of Venus is seen shown with planet-scale V-shaped cloud pattern. The image displays three ultraviolet colors and indicates a dip in the relative abundance of sulfur dioxide shown in faint blue.  Analyses of Akatsuki images and data has shown, among other discoveries, that Venus has equatorial jet similar to Earth's jet stream.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
